src/compas_model/algorithms/collider.py

```diff
@@ -0,0 +1,357 @@
+from compas.geometry import (
+    Frame,
+    Polygon,
+    Plane,
+    Transformation,
+    transform_points,
+    distance_point_point,
+    Point,
+    Vector,
+    cross_vectors,
+)
+from math import fabs
+
+try:
+    from shapely.geometry import Polygon as ShapelyPolygon
+
+    shapely_available = True
+except ImportError:
+    print("Shapely package is not available. Please install it.")
+    shapely_available = False
+
+
+def aabb_aabb(box0, box1):
+    """Verify if this axis-aligned bounding-box collides with another axis-aligned bounding-box.
+
+    Parameters
+    ----------
+    box0 : :class:`compas.geometry.Box`
+        First axis-aligned bounding-box.
+    box1 : :class:`compas.geometry.Box`
+        Second axis-aligned bounding-box.
+
+    Returns
+    -------
+    bool
+        True if the two axis-aligned bounding-box collide.
+        False otherwise.
+
+    """
+
+    p0_0 = box0.xmin, box0.ymin, box0.zmin
+    p0_1 = box0.xmax, box0.ymax, box0.zmax
+    p1_0 = box1.xmin, box1.ymin, box1.zmin
+    p1_1 = box1.xmax, box1.ymax, box1.zmax
+
+    if p0_1[0] < p1_0[0] or p1_1[0] < p0_0[0]:
+        return False
+
+    if p0_1[1] < p1_0[1] or p1_1[1] < p0_0[1]:
+        return False
+
+    if p0_1[2] < p1_0[2] or p1_1[2] < p0_0[2]:
+        return False
+
+    return True
+
+
+def box_box(box0, box1):
+    """Verify if this box collides with another box.
+
+    Parameters
+    ----------
+    box0 : :class:`compas.geometry.Box`
+        First box.
+    box1 : :class:`compas.geometry.Box`
+        Second box.
+
+    Returns
+    -------
+    bool
+        True if the two boxes collide.
+        False otherwise.
+
+    """
+    # get separation plane
+    def GetSeparatingPlane(RPos, axis, box0, box1):
```

Comment:
please use the normal conventions for function names: `get_separating_plane`

Reply:
changed